api/nodes/views.py

```diff
@@ -1104,6 +1113,25 @@ class NodeFilesList(JSONAPIBaseView, generics.ListAPIView, WaterButlerMixin, Lis
 
     ordering = ('_materialized_path',)  # default ordering
 
+    # `modified` query param would reference the FileNode's `modified` DateTime field, but the user really wants when
+    # the FileVersion (the file's content) was changed, not when the FileNode model was saved with changes.
+    mapped_sort_query_params = {
+        'modified': 'versions_modified_date',
+        '-modified': '-versions_modified_date',
+    }
```

Comment:
The most important thing is that we have a field and sortable date that are the same thing, so the list is consistent with itself. Though once we get into project files, we'll want the right date to bubble to the "most recent" when it's edited. `last_updated` sounds like a plausible field to use, but I don't know the differences between them and which is actually appropriate.

Reply:
> @jwalz I think I have a good compromise which is to to make `date_modified` into an annotation and sort on that without the mapping or extra junk, so it's always consistent with the serialization.

>. We could even consider replacing the existing date_modified in the serializer to use the annotated value so it can be used in sorting.

That's my preferred approach if there are no historical reasons that we need to keep the existing `date_modified` logid. The key here is that we need to add the annotation *everywhere* instead of letting it be `required=False` and `null=True`